js/vector.js

```diff
@@ -0,0 +1,63 @@
+class Vector {
+
+    constructor(x, y) {
+        if (arguments.length === 0) {
```

Comment:
you can just use keyword arguments here. i.e `constructor(x=0, y=0`. then you dont need the if/else block and you can just set `this.x = x;` and `this.y = y`

Reply:
til